### DISCUSSION TITLE
Zephyr summit. Miniconf related to testing .

### DISCUSSION THREAD

**Discussion by PerMac** [zephyrproject-rtos/zephyr#35845] [Ideas] [answered]
The Zephyr Summit is getting closer and I would like to discuss the topics we (people giving talks during miniconf related to testing) would like to raise and how we can share the topics efficiently between us. I was thinking about focusing during my talk on  on hardware tests and results publishing. The related topics would be:
- show how to create and use hw map
- show how to use quarantine
- result publishing process

Below I listed some other topics I think might be interesting for the audience. Is anyone interested in presenting them? Do you find them worth to be presented:
- new convention in nomenclature/test hierarchy (test case, test suite, test scenario, etc?)
- some overview of CI tests? daily plan, on-commit, various checks
- scope of CI tests? general/ test changed/ board changed.
- Integration mode (I think it is important to properly address the use of it)?

What are your plans for the talks?

**Comment by nashif** [comment, marked as answer]
We have the following 4 Talks:

- Functional Safety Verification & Validation Test Case Development and the Challenges - Steven Wang (@LeiW000 )
- Visualising Zephyr’s Health - Shen, Shihao (@s-kelvin) 
- On-target testing with Twister and the process of results publishing - Maciej Perkowski  (@PerMac )
- Twister, a Powerful Test Runner for Zephyr OS Automation Testing - Peng Chen (@chen-png)


I think the last 2 are talking about the same thing, we need to make sure we do not duplicate content. So @PerMac  and @chen-png  you need to coordinate.

**Reply by wangnuannuan** [reply]
About the Twister, can you introduce the multi-process and  multi-threading model it uses?

**Comment by galak** [comment]
A few topics:

* time it takes CI to run for a PR - how do we reduce this, for both PRs and nightly
* coverage of code from a build point of view - how to improve build testing
* do we need a sample for ever sensor?

**Reply by nashif** [reply]
> * do we need a sample for ever sensor?

i like this one, because I am working on something in the background that will fix that a bit. I do not think this is related to testing however.

**Reply by nashif** [reply]
https://github.com/zephyrproject-rtos/zephyr/pull/35854

**Reply by galak** [reply]
I agree its not 100% about testing, but goes more to CI usage, and how we define what makes sense to be added in samples.

**Reply by galak** [reply]
https://github.com/zephyrproject-rtos/zephyr/pull/34988
